tests/dotnet.buildtools.prereqs.docker.tests/ManifestTests.cs

```diff
@@ -0,0 +1,79 @@
+        //TODO: The repo root should be a runtime config value
+        string srcPath = Path.Combine(Directory.GetCurrentDirectory(), "../../../../../src");
+        var manifestFiles = Directory.GetFiles(srcPath, "manifest.json", SearchOption.AllDirectories);
+
+        foreach (var manifestFile in manifestFiles)
+        {
+            OutputHelper.WriteLine($"Processing manifest file: {manifestFile}");
+            var manifestJson = JsonDocument.Parse(File.ReadAllText(manifestFile)).RootElement;
+            var platforms = manifestJson.GetProperty("repos")
+                .EnumerateArray()
+                .SelectMany(repo => repo.GetProperty("images").EnumerateArray())
+                .SelectMany(image => image.GetProperty("platforms").EnumerateArray())
+                .ToArray();
+
+            foreach (var platform in platforms)
+            {
+                string dockerfilePath = platform.GetProperty("dockerfile").GetString() ?? string.Empty;
+                string architecture = platform.TryGetProperty("architecture", out var archProp) ? archProp.GetString()! : DefaultArchitecture;
+
+                if (dockerfilePath.Contains("/cross/"))
+                {
+                    // Check that all "cross" dockerfiles end with a folder that starts with an architecture
+                    var lastFolder = Path.GetFileName(dockerfilePath);
+                    if (!crossArchPrefixes.Any(prefix => lastFolder.StartsWith(prefix)))
+                    {
+                        invalidDockerfilePaths.Add($"Cross Dockerfile path '{dockerfilePath}' does not end with a valid architecture prefixed folder.");
+                    }
+                }
+                else
+                {
+                    // If there are multiple platform elements that reference the same Dockerfile,
+                    // then the path must not end with an architecture folder
+                    var matchingPlatforms = platforms.Where(p => p.GetProperty("dockerfile").GetString() == dockerfilePath);
+                    if (matchingPlatforms.Count() > 1)
+                    {
+                        if (dockerfilePath.EndsWith(architecture))
+                        {
+                            invalidDockerfilePaths.Add(
+                                $"Dockerfile path '{dockerfilePath}' should not end with '{architecture}' because it is built for multiple platforms.");
+                        }
+                    }
+                    else if (!dockerfilePath.EndsWith(architecture))
+                    {
+                        invalidDockerfilePaths.Add($"Dockerfile path '{dockerfilePath}' should end with an architecture folder '{architecture}'.");
+                    }
+                }
+            }
+        }
+
+        if (invalidDockerfilePaths.Any())
+        {
+            Assert.Fail($"Invalid Dockerfile paths:\n{string.Join("\n", invalidDockerfilePaths)}");
+        }
```

Comment:
Would the following give a descriptive enough error message?

```suggestion
        Assert.Empty(invalidDockerfilePaths);
```

Reply:
Good suggestion, but the UX is not what I am looking for.

      DotNet.BuildToolsPrereqs.Docker.Tests.ManifestTests.ValidateFolderStructure (235ms): Error Message: Assert.Empty() Failure: Collection was not empty
      Collection: ["Dockerfile path 'src/debian/11/helix/arm64v8' shou"···, "Dockerfile path 'src/debian/12/helix/arm32v7' shou"···, "Dockerfile path 'src/debian/12/
      helix/arm64v8' shou"···]